Update Docs - Remove Screenshots, Reorganize content, Fix style concerns (#17)
hashicorp/terraform-docs-agents: website/docs/cloud-docs/agents/requirements.mdx

---
page_title: Terraform Cloud Agents - Requirements - Terraform Cloud and Terraform Enterprise
description: >-
  Supported operating systems, Terraform versions, hardware requirements, and networking requirements for Terraform Cloud Agents.
---

> # Requirements

Reviewer: I just realized that we don't really talk about how all agents within a given agent pool _should_ have the same OS and hardware resources available. If they are not homogenous, then run performance could vary wildly. I know you are moving a lot of these docs around but maybe this would be a good time to add a quick mention of this?

Author: ahh thank you! I have added a note here to that effect - let me know what you think!

Reviewer: Perfect 👌

> 

Author: ~> **Important:** All agents in an agent pool should have the same operating system and hardware resources available. If the agents are using different resources, run performance can vary significantly between each agent in the pool.

> 

Author: ```suggestion
~> Important: All agents in an agent pool should have the same operating system and hardware resources available. If the agents are using different resources, run performance can vary significantly between each agent in the pool.

```

Ensure your system meets the following requirements before installing and configuring Terraform Cloud Agents. Refer to [Terraform Cloud Agents on TFE](/enterprise/admin/agents-on-tfe) for additional Terraform Enterprise requirements.

## Supported Operating Systems

> [Agents](https://releases.hashicorp.com/tfc-agent/) currently only support x86_64 bit Linux operating systems. You can also run the agent within Docker using our official [Terraform Agent Docker container](https://hub.docker.com/r/hashicorp/tfc-agent).

Reviewer: ```suggestion
[Agents](https://releases.hashicorp.com/tfc-agent/) currently only support x86_64 bit Linux operating systems. You can also run the agent within Docker using our official [Terraform Cloud Agent Docker container](https://hub.docker.com/r/hashicorp/tfc-agent).
```

## Supported Terraform Versions

Agents support Terraform versions 0.12 and above. Workspaces configured to use Terraform versions below 0.12 cannot select the agent-based execution mode.

## Hardware Requirements

> The host running the agent has varying resource requirements depending on the workspace. A host can be a dedicated or shared cloud instance, virtual machine, bare metal server, or a container. You should monitor and adjust memory, CPU, and disk space based on each workspace's usage and performance. The name of your instance type may vary depending on your deployment environment.

Author: I think in this case that edit might make this incorrect so in this one case I'm going to reject this. Specifically, this I believe "dedicated instance" is something folks say and so this isn't really about the user dedicating, it's about listing all of the things the host can be.

> You can use the specs below as a reference:

Author: ```suggestion
Use the following specifications as a reference:
```

- At least 4GB of free disk space
  - Each run requires the agent to temporarily store local copies of the tarred repository, extracted repository, state file, any providers or modules, and the Terraform binary itself.
- At least 2GB of system memory

## Networking Requirements

> In order for an agent to function properly, it must be able to make outbound requests over HTTPS (TCP port 443) to the Terraform Cloud application APIs. This ability may require perimeter networking as well as container host networking changes, depending on your environment. Refer to the [Terraform Cloud IP Ranges documentation](/cloud-docs/architectural-details/ip-ranges) for more details on the IP ranges.

Author: ```suggestion
In order for an agent to function properly, it must be able to make outbound requests over HTTPS (TCP port 443) to the Terraform Cloud application APIs. These requests may require perimeter networking as well as container host networking changes, depending on your environment. Refer to the [Terraform Cloud IP Ranges documentation](/cloud-docs/architectural-details/ip-ranges) for more details on the IP ranges.
```

> Additionally, the agent must also be able to communicate with any services required by the Terraform code it is executing. This includes the Terraform releases distribution service, [releases.hashicorp.com](https://releases.hashicorp.com) (supported by [Fastly](https://api.fastly.com/public-ip-list)), as well as any provider APIs. The services which run on these IP ranges are described in the table below.

Author: I'm gonna change to "The following services.." - Ah! thank you for catching this positional language! I think I just copied and pasted this into the new page - i appreciate it!
